hack/ci-e2e.sh

```diff
@@ -25,24 +28,31 @@ export PATH="${PATH}:/usr/local/go/bin"
 "${REPO_ROOT}/hack/e2e/ensure_kubectl.sh"
 
 # Build the container image with e2e tag (used in tests)
-IMG=quay.io/metal3-io/baremetal-operator:e2e make docker
-
-# Set up minikube
-minikube start --driver=kvm2
+IMG="${BMO_IMAGE_NAME}:e2e" make docker
 
 virsh -c qemu:///system net-define "${REPO_ROOT}/hack/e2e/net.xml"
 virsh -c qemu:///system net-start baremetal-e2e
-# Attach baremetal-e2e interface to minikube with specific mac.
-# This will give minikube a known reserved IP address that we can use for Ironic
-virsh -c qemu:///system attach-interface --domain minikube --mac="52:54:00:6c:3c:01" \
-  --model virtio --source baremetal-e2e --type network --config
 
-# Restart minikube to apply the changes
-minikube stop
-minikube start
+export KUBECONFIG_E2E="/tmp/kubeconfig-bmo-e2e"
+export KUBECONFIG_E2E_UPGRADE="/tmp/kubeconfig-bmo-e2e-upgrade"
+
+# Set up minikube
+for profile in "bmo-e2e" "bmo-e2e-upgrade"; do
+  export KUBECONFIG="/tmp/kubeconfig-${profile}"
+  minikube start --profile $profile --driver=kvm2
+
+  # Attach baremetal-e2e interface to minikube with specific mac.
+  # This will give minikube a known reserved IP address that we can use for Ironic
+  virsh -c qemu:///system attach-interface --domain "${profile}" --mac="52:54:00:6c:3c:01" \
```

Comment:
Hmm. In my understanding we never have an BMH in both clusters, so this should be ok.

Reply:
That may be true for now, but it is not something we can count on in the future. I'm also afraid that this will make the connection to Ironic quite unstable. If both minikubes have the same IP and Mac then probably BMO will send requests to both of them. One is working and the other is just sitting idle without Ironic even running. So BMO would get connection refused every time a request goes to the wrong minikube